modules/postgres.nix

```diff
@@ -245,7 +245,7 @@ in
 
             # install/update pg_stat_statements extension in all databases
             # based on https://git.catgirl.cloud/999eagle/dotfiles-nix/-/blob/main/modules/system/server/postgres/default.nix#L294-302
-            (lib.mkIf cfg.configurePgStatStatements (lib.concatStrings (map (db:
+            (lib.mkIf (cfg.settings.shared_preload_libraries != null) (lib.concatStrings (map (db:
```

Comment:
This needs to be changed to depend on a new option so that we do not change any defaults. If that new option is flase but cfg.configurePgStatStatements is true, it only loads the pg_stat_statements extension.